src/reactor.cc

```diff
@@ -158,10 +175,25 @@ std::set<cyclus::RequestPortfolio<Material>::Ptr> Reactor::GetMatlRequests() {
   std::set<RequestPortfolio<Material>::Ptr> ports;
   Material::Ptr m;
 
-  int n_assem_order =
-      n_assem_core - core.count() + n_assem_fresh - fresh.count();
+  // second min expression reduces assembles to amount needed until
+  // retirement if it is near.
+  int n_assem_order = n_assem_core - core.count() + n_assem_fresh - fresh.count();
+
+  if (exit_time() != -1) {
+    int tleft = exit_time() - context()->time();
+    int tleftcycle = cycle_time + refuel_time - cycle_step;
+    double ncyclesleft = (double)(tleft - tleftcycle) / (double)(cycle_time + refuel_time);
```

Comment:
need static casts here and below

https://google-styleguide.googlecode.com/svn/trunk/cppguide.html#Casting